src/plugins/graphql.js

```diff
@@ -172,36 +217,49 @@ function normalizeArgs (args) {
   }
 }
 
-function createOperationSpan (tracer, config, operation, source) {
+function createOperationSpan (tracer, config, operation, source, variableValues, startTime) {
   const type = operation.operation
   const name = operation.name && operation.name.value
-
   const parentScope = tracer.scopeManager().active()
-  const span = tracer.startSpan(`graphql.${operation.operation}`, {
-    childOf: parentScope && parentScope.span(),
-    tags: {
-      'service.name': getService(tracer, config),
-      'resource.name': [type, name].filter(val => val).join(' '),
-      'graphql.document': source
+  const tags = {
+    'service.name': getService(tracer, config),
+    'resource.name': [type, name].filter(val => val).join(' '),
+    'graphql.document': source
+  }
+  if (variableValues && config.variables) {
```

Comment:
Just to be safe, I'd add a check that `config.variables` is actually a function. This will avoid a misconfiguration from throwing an exception. An additional check could be done once in the beginning to log an error as well if this is the case.

Reply:
I agree that helping in configuration is a good thing, so I’ll perform that check as early as possible. But what would you like it to do in the case that it’s incorrect, throw an error?

Reply:
If it's incorrect, the configuration should simply be ignored and a log entry should be created.